src/analysis/retail/evoker/preservation/modules/core/MasteryEffectiveness.tsx

```diff
@@ -115,84 +88,68 @@ class MasteryEffectiveness extends Analyzer {
   }
 
   onTwinGuardianShieldApplication(event: ApplyBuffEvent | RefreshBuffEvent) {
+    this.totalShieldEvents += 1;
+    let existingAbsorb = 0;
+    //shield refresh event's absorb value is a sum of any exising absorb and the new absorb. find the existing absorb for calc in map then remove
     if (this.twinGuardianShieldApplications.get(event.targetID)) {
+      existingAbsorb = this.twinGuardianShieldApplications.get(event.targetID)?.event.absorb || 0;
       this.twinGuardianShieldApplications.set(event.targetID, null);
     }
-
-    this.totalPossibleAbsorbs += event.absorb || 0;
-    this.totalCalculatedBaseAbsorbs += this.calculateShield(event.ability.guid);
-
+    //check for mastery map
+    const masteryEvent = this.masteryBoostCheck(event, existingAbsorb);
+    if (masteryEvent) {
+      debug && console.log('mastery boosted TG found');
+      this.totalShieldEventsAffectedByMastery += 1;
+    }
+    //push new shield to map
     this.twinGuardianShieldApplications.set(event.targetID, {
       event: event,
-      baseShieldValue: this.calculateShield(event.ability.guid),
-      hasMasteryBoost: this.masteryBoostCheck(event),
+      baseShieldValue: this.calculateShield(event, existingAbsorb),
+      hasMasteryBoost: masteryEvent,
       masteryValue: this.statTracker.currentMasteryPercentage,
-      healing: 0,
     });
   }
 
   onAnomalyShieldApplication(event: ApplyBuffEvent | RefreshBuffEvent) {
+    this.totalShieldEvents += 1;
+    let existingAbsorb = 0;
+    //shield refresh event's absorb value is a sum of any exising absorb and the new absorb. find the existing absorb for calc in map then  remove
     if (this.anomalyShieldApplications.get(event.targetID)) {
+      existingAbsorb = this.anomalyShieldApplications.get(event.targetID)?.event.absorb || 0;
```

Comment:
note: this won't handle cases where the value is reduced by ticking damage between shield applications/refreshes. not sure if that matters for this case, though